Add timezone version reporting to configure step in test runs

robert-jenner opened the pull request:
#### 2dbc4af35c1dd74b5d5276e05e8094483f31165b
<pre>
Add timezone version reporting to configure step in test runs
<a href="https://bugs.webkit.org/show_bug.cgi?id=251556">https://bugs.webkit.org/show_bug.cgi?id=251556</a>
rdar://problem/104437369

Reviewed by NOBODY (OOPS!).

This is to add the timezone database version to our configure logs so we could easily track regressions based on timezone changes.

* Tools/CISupport/build-webkit-org/steps.py:
(PrintConfiguration):
* Tools/CISupport/ews-build/steps.py:
(PrintConfiguration):
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/2dbc4af35c1dd74b5d5276e05e8094483f31165b

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/106143 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/15193 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/38975 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/115324 "Built successfully") | [  ~~🛠 🧪 win~~](https://ews-build.webkit.org/#/builders/10/builds/175398 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/110046 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/16626 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/6370 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/98346 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/115017 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/111900 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/12654 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/95623 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/35/builds/40192 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/94527 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/27266 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/81876 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/8442 "Built successfully") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/70/builds/28618 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/8939 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/84/builds/5169 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [❌ 🧪 services](https://ews-build.webkit.org/#/builders/20/builds/104950 "Failed build.webkit.org unit tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/14557 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/48163 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/10479 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

ryanhaddad commented on Tools/CISupport/build-webkit-org/steps.py:
```diff
@@ -1262,7 +1262,7 @@ class PrintConfiguration(steps.ShellSequence):
     logEnviron = False
     command_list_generic = [['hostname']]
-    command_list_apple = [['df', '-hl'], ['date'], ['sw_vers'], ['system_profiler', 'SPSoftwareDataType', 'SPHardwareDataType'], ['xcodebuild', '-sdk', '-version']]
+    command_list_apple = [['df', '-hl'], ['date'], ['sw_vers'], ['system_profiler', 'SPSoftwareDataType', 'SPHardwareDataType'], ['xcodebuild', '-sdk', '-version'], ['cat /usr/share/zoneinfo/+VERSION']]
```
I think this will need to be `['cat', '/usr/share/zoneinfo/+VERSION']`, but we might want to find a way to output something about what this string is.

I know one could do something like `echo TimeZone Data Version: $(cat /usr/share/zoneinfo/+VERSION)` to get output like `TimeZone Data Version: <blah>`, but I'm not sure if that will directly translate to the way commands are being run in this step.

robert-jenner replied:
Yeah. I was thinking about that. This was the very first draft just to see what it would do. I'll keep playing around with it.

JonWBedard commented on Tools/CISupport/build-webkit-org/steps.py:
```diff
@@ -1262,7 +1262,7 @@ class PrintConfiguration(steps.ShellSequence):
     logEnviron = False
     command_list_generic = [['hostname']]
-    command_list_apple = [['df', '-hl'], ['date'], ['sw_vers'], ['system_profiler', 'SPSoftwareDataType', 'SPHardwareDataType'], ['xcodebuild', '-sdk', '-version']]
+    command_list_apple = [['df', '-hl'], ['date'], ['sw_vers'], ['system_profiler', 'SPSoftwareDataType', 'echo TimezoneVers: $(cat /usr/share/zoneinfo/+VERSION)', 'SPHardwareDataType'], ['xcodebuild', '-sdk', '-version']]
```
The is a list of commands, each sub-list is the command argument. What you're trying to do is basically add `['echo', 'TimezoneVers: $(cat /usr/share/zoneinfo/+VERSION)']`. We need to test this on the staging instance, though, because `$(cat ...)` bit indicates you're trying to use shell scripting, so you probably need to do `['sh', 'echo TimezoneVers: $(cat /usr/share/zoneinfo/+VERSION)']` so that you are invoke with the shell instead of just subprocess.

aj062 commented on Tools/CISupport/ews-build/steps_unittest.py:
```diff
@@ -5044,7 +5044,8 @@ def test_failure(self):
     func(fullname, *argrest)
 OSError: [Errno 2] No such file or directory'''),
-            ExpectShell(command=['system_profiler', 'SPSoftwareDataType', 'SPHardwareDataType'], workdir='wkdir', timeout=60, logEnviron=False) + 0,
+            ExpectShell(command=['system_profiler', 'SPHardwareDataType'], workdir='wkdir', timeout=60, logEnviron=False) + 0,
```
Seems like removal of `SPSoftwareDataType` here was unintentional (and causing ews unit-test failure).

aj062 commented on Tools/CISupport/build-webkit-org/steps_unittest.py:
```diff
@@ -1284,7 +1285,7 @@ def test_failure(self):
     func(fullname, *argrest)
 OSError: [Errno 2] No such file or directory'''),
-            ExpectShell(command=['system_profiler', 'SPSoftwareDataType', 'SPHardwareDataType'], workdir='wkdir', timeout=60, logEnviron=False) + 0
+            ExpectShell(command=['system_profiler', 'SPSoftwareDataType', 'SPHardwareDataType'], ['sh', 'echo TimezoneVers: $(cat /usr/share/zoneinfo/+VERSION)'], workdir='wkdir', timeout=60, logEnviron=False) + 0
```
Seems like you need to add a separate `ExpectShell` line for timezone related command.

robert-jenner committed:
Add timezone version reporting to configure step in test runs
https://bugs.webkit.org/show_bug.cgi?id=251556
rdar://problem/104437369

Reviewed by NOBODY (OOPS!).

This is to add the timezone database version to our configure logs so we could easily track regressions based on timezone changes.

* Tools/CISupport/build-webkit-org/steps.py:
(PrintConfiguration):
* Tools/CISupport/ews-build/steps.py:
(PrintConfiguration):

JonWBedard commented on Tools/CISupport/build-webkit-org/steps.py:
```diff
@@ -1262,7 +1262,7 @@ class PrintConfiguration(steps.ShellSequence):
     logEnviron = False
     command_list_generic = [['hostname']]
-    command_list_apple = [['df', '-hl'], ['date'], ['sw_vers'], ['system_profiler', 'SPSoftwareDataType', 'SPHardwareDataType'], ['xcodebuild', '-sdk', '-version']]
+    command_list_apple = [['df', '-hl'], ['date'], ['sw_vers'], ['system_profiler', 'SPSoftwareDataType', 'SPHardwareDataType'], ['echo TimezoneVers: $(cat /usr/share/zoneinfo/+VERSION)'], ['xcodebuild', '-sdk', '-version']]
```
Should be [`/bin/sh', '-c', 'echo TimezoneVers: $(cat /usr/share/zoneinfo/+VERSION)']
